### PR TITLE
Refine config.md

### PR DESCRIPTION
**Pull request by pellared**
## Why

`config.md` maintainace cleanup.

## What

- Refine the .NET CLR Profiler section - I tried to describe when it is not needed and when it is required
- Use .NET (Core) instead of .NET Core were applicable
- Restructure logs instrumentations docs
- Table format
- Change the "instrumentation libraries" headers

### CODE REVIEW

**Comment by pellared** [comment]
@rajkumar-rangaraj PTAL at the "Logs instrumentations" section and make sure I am not missing anything important (or lying)

File: docs/config.md [hunk 1]
```diff
@@ -2,85 +2,69 @@
...
+| `AspNet`     | ASP.NET Core                                                                                                                                                                                    | *                  | source               |
+| `HttpClient` | [System.Net.Http.HttpClient](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient) and [System.Net.HttpWebRequest](https://docs.microsoft.com/dotnet/api/system.net.httpwebrequest) | *                  | source               |
+| `NetRuntime` | [OpenTelemetry.Instrumentation.Runtime](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Runtime)                                                                                   | *                  | source               |
+
+### Logs instrumentations
+
+| ID               | Instrumented library                                                                        | Supported versions | Instrumentation type   |
```
Comment (rajkumar-rangaraj) [inline docs/config.md:60]: `ID` is misleading here, should we remove `ID` column until we get the support for other logging provider and environment variables to support it.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM, left a minor suggestion.

File: docs/config.md (hunk 1 above)
Comment (pellared) [reply docs/config.md]: Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1311/commits/b9dc3c4ed2da4f54db716a033a3db96c90de2405

**Review by nrcventura — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

File: docs/config.md [hunk 2]
```diff
@@ -219,32 +203,37 @@ OpenTelemetry .NET Automatic Instrumentation.
 
 ## .NET CLR Profiler
 
-To perform bytecode instrumentation, configure the OpenTelemetry .NET
-Automatic Instrumentation as a .NET CLR Profiler. The CLR uses the following
+Setting OpenTelemetry .NET Automatic Instrumentation as a .NET CLR Profiler
+is always required for .NET Framework.
+
+On .NET (Core), the .NET CLR Profiler is used only to perform bytecode instrumentation.
```
Comment (RassK) [inline docs/config.md:209]: I think the CLR Profiler ILogger integration is also going to this section?

**Pull request merged**